File: LeetCode/2924-find-champion-ii/2924-find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int[] indegree = new int[n];
        for (int[] x : edges) {
            indegree[x[1]]++;
        }
        int ans = -1;
        int freq = 0;
        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) {
                ans = i;
                freq++;
                if (freq > 1) return -1;
            }
        }
        return ans;
    }
}

File: LeetCode/2924-find-champion-ii/2924-find-champion-ii.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        indegree = [0] * n
        for x in edges:
            indegree[x[1]] += 1
        ans, freq = -1, 0
        for i in range(n):
            if indegree[i] == 0:
                ans = i
                freq += 1
                if freq > 1:
                    return -1
        return ans